### DISCUSSION TITLE
wrong generation of noah mp cdf file

### DISCUSSION THREAD

**Discussion by der-knight** [NASA-LIS/LISF#831] [Q&A] [answered]
hello lisf community,

I need help with generating cdf file for OL simulation of noah mp 3.6. The OL simulation shows a good correlation with ground data and now I want to see how Data assimilation of smops will affect the output. The code runs fine but produces a blank output.
Please find attached the log file, config file and the output nc file in the zipped folder. 
[New folder.zip](https://github.com/NASA-LIS/LISF/files/6796422/New.folder.zip)

**Comment by dmocko** [comment, marked as answer]
Hello,

Thanks for reaching out, and for including the attachments.

Looking at your ldtlog.0000 file, I see that LDT is unable to find your LIS files:

[INFO] LDT cycle time: 03/02/2008 00:00:00
 [WARN] LIS file ./OL_OUTPUT/noahmp/SURFACEMODEL/200803/LIS_HIST_200803020000.d0
 0.nc
 [WARN] not found ...

This warning message continues for every single daily LIS file.

Your ldt.config file includes the following config entry:

LIS soil moisture output directory:        ./OL_OUTPUT/noahmp

Are you sure that the "OL_OUTPUT" directory is found in the directory where are you are running LDT?

What does this command show for you, in the same directory as where you are running LDT?

ls -l ./OL_OUTPUT/noahmp/SURFACEMODEL/200803/LIS_HIST_200803020000.d00.nc

David

**Reply by der-knight** [reply]
Hi David,
Thank you for your reply
I am sorry I was troubleshooting this problem and by mistake provided the wrong log file. I am in the right directory, and ldt is able to read the OL outputs.
PFA the original log file and config file
[New folder.zip](https://github.com/NASA-LIS/LISF/files/6805303/New.folder.zip)

**Reply by dmocko** [reply]
Hello,

Thanks for attaching the new files.

I am unable to determine why you are seeing zero outputs for your run.

Can you please confirm that you have been able to successfully run step4 of our end-to-end testcase?
https://lis.gsfc.nasa.gov/tests/lis

If you are able to generate the CDFs in step4 for the domain in the end-to-end testcase, then we can try to figure out why the CDFs have zero outputs for your domain.

If you have successfully generated the CDFs for step 4, then please also attach a sample LIS_HIST file as well as the ldt_param_noahmp25.nc file.

David

**Reply by der-knight** [reply]
Hello David,
I have completed the test case and it runs fine. I  understand that the only difference between Noah
 3.6 and noahmp 3.6 is the pbl height variable and that is the only change i made. 
 Please find attached lis hist file, parameter file and the soil moisture graph of OL simulation and ismn data.
P.S I have multiple layer soil moisture and I want to generate cdf only for the top layer data assimilation in accordance with satellite soil moisture but I could not find configuration to specify that
[files.zip](https://github.com/NASA-LIS/LISF/files/6812771/files.zip)

**Reply by dmocko** [reply]
Thanks, I will look at your files tomorrow.  I believe that the CDFs are generated for all layers of the modeled soil moisture, as the cross correlations in the DA perturbation attributes file will perturb all soil moisture layers, even though only the top layer is assimilated from the satellite soil moisture.

**Reply by dmocko** [reply]
Hello,

Thanks for attaching these files.  From looking at them, I am able to identify a likely cause of your zero CDFs.

LDT assumes that the LIS_HIST output files includes soil moisture in units of volumetric [m^3 m-3].  Your LIS_HIST output has soil moisture in [kg m-2].

Here is the code where LIS is looking for the data in volumetric:  LISF/ldt/DAobs/LISlsmSMobs/LISlsmSM_obsMod.F90

!  which variable we want in the DA obs computations. 
    call LDT_initializeDAobsEntry(LDT_DAobsData(1)%soilmoist_obs, "m3/m3",1,1)

I wasn't aware of this requirement, and after looking through our documents and Users' Guide, we don't specify this.  We will correct our documentation.

I would recommend that you re-run the LIS Open Loop simulation and change the MODEL OUTPUT TBL to write the soil moisture in volumetric.  The end-to-end testcase writes/reads the soil moisture in volumetric, which is why your testcase run produced the correct CDF values.  After you re-run LIS to output soil moisture in volumetric for your domain, run LDT again, and you should see good values in the CDF files for your domain.

I don't believe that it's advised to change the above code to try to read your existing LIS_HIST output data in "kg/m2", as the observations are also assumed to be in "m3/m3" volumetric.  Any comments, @LIS-navari or @yhkwon81 ?

Apologies for the trouble,

David

**Reply by der-knight** [reply]
Hello David,
Thank you for your help
Your diagnosis was correct; on changing the soil moisture to volumetric correct CDFs are generated, and the DA run also works fine

Regards
nirdesh
